### PR TITLE
Throw exception if exceeds 80w without paging

### PR DESCRIPTION
**Pull request by zhoney**
resolved: #360

Change-Id: I46c7a61df21d6902aba6c0c965e8f63aa3b76fe2

### CODE REVIEW

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/VerticesAPI.java [hunk 1]
```diff
@@ -111,10 +111,13 @@ public String scan(@Context GraphManager manager,
 
         ConditionQuery query = new ConditionQuery(HugeType.VERTEX);
         query.scan(start, end);
-        query.limit(Query.DEFAULT_CAPACITY);
         query.page(page);
+        boolean paging = page != null;
+        if (paging) {
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/VerticesAPI.java]: use query.paging() instead

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/EdgesAPI.java [hunk 2]
```diff
@@ -111,10 +111,13 @@ public String scan(@Context GraphManager manager,
 
         ConditionQuery query = new ConditionQuery(HugeType.EDGE_OUT);
         query.scan(start, end);
-        query.limit(Query.DEFAULT_CAPACITY);
         query.page(page);
+        boolean paging = page != null;
+        if (paging) {
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/EdgesAPI.java]: use query.paging() instead

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/EdgesAPI.java [hunk 3]
```diff
@@ -111,10 +111,12 @@ public String scan(@Context GraphManager manager,
 
         ConditionQuery query = new ConditionQuery(HugeType.EDGE_OUT);
         query.scan(start, end);
-        query.limit(Query.DEFAULT_CAPACITY);
         query.page(page);
+        if (query.paging()) {
+            query.limit(Query.DEFAULT_CAPACITY);
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/EdgesAPI.java]: set a more reasonable value to balance performance and timeout risk

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/API.java [hunk 4]
```diff
@@ -48,6 +48,9 @@
 
     private static final Logger LOG = Log.logger(RestServer.class);
 
+    // Empirical value of scan limit, with which results can be returned in 3s
+    protected static final long SCAN_LIMIT = 100_000L;
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/API.java]: SCAN_PAGE_LIMIT

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/EdgesAPI.java [hunk 5]
```diff
@@ -112,7 +112,7 @@ public String scan(@Context GraphManager manager,
         query.scan(start, end);
         query.page(page);
         if (query.paging()) {
-            query.limit(SCAN_LIMIT);
+            query.limit(SCAN_PAGE_LIMIT);
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/EdgesAPI.java]: add page_limit parameter and set default value as SCAN_PAGE_LIMIT
rename SCAN_PAGE_LIMIT to SCAN_DEFAULT_PAGE_LIMIT

**Comment by zhoney** [comment]
abandon due to included in #428

**Pull request closed without merging**